### DISCUSSION TITLE
Migration guide for TeX users/programmers?

### DISCUSSION THREAD

**Discussion by user202729** [typst/typst#307] [General]
While a niche topic, it might occasionally be helpful.

For example, I believe 

TeX | Typst 
---|---
hbox | box/text
vbox | block
halign | grid/align
leader | repeat
rotatebox | rotate
scalebox | scale
raisebox | move
phantom | hide
smash | (can't find?)
ht/dp/wd | measure

Side note, I can't implement `smash` with `box` (probably because I still don't fully understand how the boxes in Typst work),
and can only implement `llap` in a somewhat hacky way:

```
#let rlap(content) = box(width: 0pt, content)
#let llap(content) = box(width: 0pt)[#h(-100cm)#h(1fr)cd]
```

Maybe consider giving some warning on overfull (h)box like LaTeX does?

For math there's already #162.

**Comment by jason-s** [comment]
or: (Tex on left, Typst on right)

- `varepsilon` <-> `epsilon`
- `epsilon` <-> `epsilon.alt`
- `varphi` <-> `phi`
- `phi` <-> `phi.alt`
- `bar` <-> `macron`

**Comment by hyrious** [comment]
The $\TeX=$

![E61093518189A452E9CE3F45BA49D81A](https://user-images.githubusercontent.com/8097890/234753603-3b423f1f-6ee8-45e0-abc6-ecc51a60a9c9.jpg)

**Comment by fabianbosshard** [comment]
Somewhat related:
 Is it be possible to do something like this?
```
\newcommand*\matrspace{0.8mu}
\newcommand{\matr}[1]{\mspace{\matrspace}\underline{\mspace{-\matrspace}\smash[b]{#1}\mspace{-\matrspace}}\mspace{\matrspace}}
```
which, imho, looks nicer than a raw underline, if there is e.g. a product of matrices, or symbols with different depth:
```
\[
\matr{A} \matr{v} = \matr{Q} \matr{R} \matr{v}
\]
\[
\underline{A} \underline{v} = \underline{Q} \underline{R} \underline{v}
\]
```
<img width="229" height="155" alt="image" src="https://github.com/user-attachments/assets/775b843c-b607-4b0f-8ae9-493d9a01acc3" />

**Reply by fabianbosshard** [reply]
Update:

I don't know if it is possible to reproduce LaTeX-style `\smash` purely with existing typst methods, but I added a small smash-analogue to the source code (I called it `squish`, since `smash` is already the name of a symbol and actually `squish` makes more sense IMHO, since the resulting object is _squished_ together, not _smashed_ into pieces).

Then it is possible to achieve the same thing, even though it seems more complicated than in LaTeX:
```
#let matr(x) = context {
  let em_abs = measure(math.inline[box(width: 1em)]).width.to-absolute()
  let mu = em_abs / 18
  let matrspace = 0.1 * mu

  math.class("normal",
    math.squish(h(matrspace)
    + math.underline(
        math.squish(h(-matrspace) + math.bold(x) + h(-matrspace), mode: "bottom")
      )
    + h(matrspace), mode: "bottom")
  )
}
```
I don't know why, but without the "freezing" of the font size, the amount subtracted looks slightly wrong for script and for sscript it is even worse. Anyway, this "makro" can be called in a simple way:
```
$ matr(A) = matr(Q) matr(R) bold(x) $
$ underline(bold(A)) = underline(bold(Q)) underline(bold(R)) bold(x) $
$ bold(A) = bold(Q) bold(R) bold(x) $
```
<img width="551" height="569" alt="image" src="https://github.com/user-attachments/assets/c0a2ab6b-3748-4f6e-9b27-c9c15c6bc140" />


The binary I used for the above, has been compiled from my fork, specifically from
[this branch](https://github.com/fabianbosshard/typst/tree/smash) and [this commit](https://github.com/fabianbosshard/typst/commit/7d11f750399f2286a0653fc2ec07e32476ced93d).
The code is prototypish and was mostly produced by Codex, so I don't know about the quality of it.
if I have time I will try to clean it and check if everything makes sense.
